numpy/core/src/multiarray/buffer.c

```diff
@@ -797,7 +786,17 @@ array_getbuffer(PyObject *obj, Py_buffer *view, int flags)
     view->buf = PyArray_DATA(self);
     view->suboffsets = NULL;
     view->itemsize = PyArray_ITEMSIZE(self);
-    view->readonly = !PyArray_ISWRITEABLE(self);
+    /*
+	 * If a read-only buffer is requested on a read-write array, we return a
+	 * read-write buffer as per buffer protocol.
```

Comment:
So to be clear, the behavior was not actually dubious as previously remarked, but is specified by the buffer protocol?

Reply:
I think the dubious part is either: The buffer protocol should be doing it differently, or, the deprecation side of things is dubious (since this deprecation was used before for the diagonal stuff).

Reply:
Note the incorrect indent. Tabs?

Reply:
oops, wow emberassing, I wonder which of my editors has such an insane setup :).

Reply:
Here's the text I was looking for:

> Controls the readonly field. If set, the exporter MUST provide a writable buffer or else report failure. Otherwise, the exporter MAY provide either a read-only or writable buffer, but the choice MUST be consistent for all consumers.

Is _"the choice MUST be consistent for all consumers."_ relevant here? I'm struggling to attach meaning to it.